tools/bin/commands/link.ts

```diff
@@ -1,175 +1,133 @@
-    return false;
-  }
-
-  for (const p of Object.values(paths)) {
-    if (!fs.existsSync(p)) {
-      color.error(`Directory ${p} not found. Run the command from the public package repository`);
-      return false;
-    }
-  }
-
-  const dependencies = readDependencies(curDir);
-
-  if (local) {
-    const packageHierarchy: string[] = getHierarchy(curDir);
-    function link(packagePath: string): void {
-      const packageJsonPath = path.join(packagePath, 'package.json');
-      const json = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
-      json.scripts['link-all'] = generateLinkScript(packagePath, packageHierarchy);
-      json.scripts['build-all'] = generateBuildScript(packagePath, packageHierarchy);
-      fs.writeFileSync(packageJsonPath, JSON.stringify(json, null, 2), 'utf-8');
-      exec('npm install && npm link && npm run link-all', {cwd: packagePath}, (err, stdout, stderr) => {
-        if (err) throw err;
-        else console.log(stderr, stdout);
-      });
-    }
+let devMode = false;
+let repositoryDir = curDir;
+let localPackageDependencies: PackageData[];
+let packagesToLink: Set<string>;
+
+while (repositoryDir.length > 0) {
+  if (repositoryDirNameRegex.test(repositoryDir))
+    break;
+  repositoryDir = path.dirname(repositoryDir);
+}
 
-    packageHierarchy.forEach((packageName) => link(getPackagePath(packageName)));
-    link(curDir);
-  } else {
-    runScript(curDir, 'npm install', dependencies, {
-      dirMessage: 'Unlinking local packages in ',
-      successMessage: 'Local packages have been successfully unlinked.',
-    });
-  }
+const apiDir = path.join(repositoryDir, 'js-api');
+const libDir = path.join(repositoryDir, 'libraries');
+const packageDir = path.join(repositoryDir, 'packages');
 
+const apiPackageName = 'datagrok-api';
+const libName = '@datagrok-libraries/';
+const packageName = '@datagrok/';
+
+export async function link(args: LinkArgs) {
+  localPackageDependencies = []
+  packagesToLink = new Set<string>();
+  if (args.dev !== undefined)
+    devMode = args.dev;
+  collectPackagesData(curDir);
+  await linkPackages();
   return true;
 }
 
-function readDependencies(packagePath: string): utils.Indexable {
-  if (packagePath in packageDependencies)
-    return packageDependencies[packagePath];
-  const fileContent = fs.readFileSync(path.join(packagePath, 'package.json'), 'utf-8');
-  const json = JSON.parse(fileContent);
-  const libs: utils.Indexable = {};
-  for (const dep in json.dependencies) {
-    if (dep === apiPackageName || dep.startsWith(`${libScope}/`))
-      libs[dep] = json.dependencies[dep];
+function collectPackagesData(packagePath: string = curDir): string[] {
+  const packageJsonPath = path.join(packagePath, 'package.json');
+  const json = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+  let result: string[] = [];
+  let dependencies = json.dependencies;
+  if (devMode)
+    dependencies = json.devDependencies;
```

Comment:
I think it would be much better to link both in this case. I don't see a use case for linking only dev dependencies